### PR TITLE
Add tests to check if SonarQubeNavigationItem is free-threaded

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Part of #4856

### CODE REVIEW

File: src/Integration.TeamExplorer.UnitTests/SonarQubeNavigationItemTests.cs [hunk 1]
```diff
@@ -30,29 +32,49 @@ namespace SonarLint.VisualStudio.Integration.UnitTests.TeamExplorer
     [TestClass]
     public class SonarQubeNavigationItemTests
     {
+        [TestMethod]
+        [Ignore] // Object is created successfully, but an exception when
```
Comment (duncanp-sonar) [inline src/Integration.TeamExplorer.UnitTests/SonarQubeNavigationItemTests.cs:36]: I'll create a separate ticket for this and fix it separately.

File: src/Integration.TeamExplorer/SonarQubeNavigationItem.cs [hunk 2]
```diff
@@ -49,6 +50,12 @@ internal SonarQubeNavigationItem([Import] ITeamExplorerController controller)
             this.IsVisible = true;
             this.IsEnabled = true;
 
+            // Note on threading:
+            // MEF constructors must be free-threaded i.e. capable of running to completion the calling thread.
+            // There's no public documentation on how these whether it's ok to create WPF artefacts like brushes
+            // and icons in the constructor. However, this is what the VS Team Explorer implementations of this
+            // class are doing, so we assume it's ok.
+            // See [VS installation directory]\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer\Microsoft.TeamFoundation.TeamExplorer.Navigation.dll
```
Comment (duncanp-sonar) [inline src/Integration.TeamExplorer/SonarQubeNavigationItem.cs:58]: It wasn't obvious whether our code was ok or not, so I found and decompiled the VS implementations to check what they did.

File: src/Integration.TeamExplorer/SonarQubeNavigationItem.cs [hunk 3]
```diff
@@ -28,6 +28,7 @@
 namespace SonarLint.VisualStudio.Integration.TeamExplorer
 {
     [TeamExplorerNavigationItem(SonarQubeNavigationItem.ItemId, SonarQubeNavigationItem.Priority, TargetPageId = SonarQubePage.PageId)]
+    [PartCreationPolicy(CreationPolicy.NonShared)] // The VS navigations in MS.TeamFoundation.TeamExplorer.Navigation are non-shared
```
Comment (duncanp-sonar) [inline src/Integration.TeamExplorer/SonarQubeNavigationItem.cs:31]: The VS components are marked as non-shared.

**Review by bigfluffycookie — APPROVED** [review]
Nice

**Pull request merged**